crates/engine/tree/src/tree/cached_state.rs

```diff
@@ -467,8 +471,13 @@ impl ExecutionCache {
                 continue
             }
 
-            // If the account was destroyed, invalidate from the account / storage caches
-            if account.was_destroyed() {
+            // If the account was destroyed, invalidate from the account / storage caches.
+            //
+            // Post-cancun when EIP-6780 is live, an account can be destroyed only when it's created
+            // in the same transaction. This guarantees that we will not have such accounts
+            // and storage slots in our cache, because Revm doesn't go through the
+            // Database for freshly created accounts. Hence we can safely ignore invalidating them.
```

Comment:
this doesnt mention state, can we elaborate why this is okay to not invalidate state